.circleci/scripts/vs_install.ps1

```diff
@@ -1,15 +1,7 @@
-$VS_DOWNLOAD_LINK = "https://aka.ms/vs/15/release/vs_buildtools.exe"
+$VS_DOWNLOAD_LINK = "https://aka.ms/vs/16/release/vs_community.exe"
 $COLLECT_DOWNLOAD_LINK = "https://aka.ms/vscollect.exe"
-$VS_INSTALL_ARGS = @("--nocache","--quiet","--wait", "--add Microsoft.VisualStudio.Workload.VCTools",
-                                                     "--add Microsoft.VisualStudio.Component.VC.Tools.14.13",
-                                                     "--add Microsoft.Component.MSBuild",
-                                                     "--add Microsoft.VisualStudio.Component.Roslyn.Compiler",
-                                                     "--add Microsoft.VisualStudio.Component.TextTemplating",
-                                                     "--add Microsoft.VisualStudio.Component.VC.CoreIde",
-                                                     "--add Microsoft.VisualStudio.Component.VC.Redist.14.Latest",
-                                                     "--add Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Core",
-                                                     "--add Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
-                                                     "--add Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Win81")
+$VS_INSTALL_ARGS = @("modify","--nocache","--quiet","--wait", "--installPath ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Community""",
+                                                              "--add Microsoft.VisualStudio.Component.VC.14.27.x86.x64")
```

Comment:
I tested that VC.14.27.x86.x64 works, but VC.14.27.29110.x86.x64 doesn't work.
![image](https://user-images.githubusercontent.com/16190118/112464882-0188ef80-8d9f-11eb-9a49-57bdb70cfd69.png)

Reply:
It won't. the full list of the installable components is written [here](https://docs.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-build-tools?view=vs-2019). How could you install a non-existent one?

Reply:
if we couldn't specify the build number, the vc version isn't pinned in fact.

Reply:
for example, 14.28.29333 (VS 16.8.5)works , 14.28.29910 (VS 16.9.1)not.

Reply:
@mszhanyi Wow, will this actually happen? If this happens, then I think the versioning mechanism should be improved since when using Linux, we will only say we need GCC 5.4, but not GCC 5.4.1.

Reply:
Looks like it really happens, since in your PR, you use 14.28.29333. OMG, that's unbelievable.

Reply:
I guess there is one way we could achieve this. First, let's download an old installer like what you did (current ones for example). If they won't update it themselves, then we know 14.28 refers to the old one we want.

Reply:
And in my impression, those failing complex tests passed in 14.28.29333.

Reply:
@mszhanyi Well, have to say MSVC is still too inmature. How could those things happen in serious things like compilers? As far as I know, `std::acos<std::complex>` is in C++11.